functional/unit_action_test.go

```diff
@@ -224,3 +226,68 @@ func TestUnitSSHActions(t *testing.T) {
 		t.Errorf("Could not find expected string in journal output:\n%s", stdout)
 	}
 }
+
+// copyFile()
+func copyFile(newFile, oldFile string) error {
+	input, err := ioutil.ReadFile(oldFile)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(newFile, []byte(input), 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// genNewFleetService() is a helper for generating a temporary fleet service
+// that reads from oldFile, replaces oldVal with newVal, and stores the result
+// to newFile.
+func genNewFleetService(newFile, oldFile, newVal, oldVal string) error {
+	input, err := ioutil.ReadFile(oldFile)
+	if err != nil {
+		return err
+	}
+	lines := strings.Split(string(input), "\n")
+
+	for i, line := range lines {
+		if strings.Contains(line, oldVal) {
+			lines[i] = strings.Replace(line, oldVal, newVal, len(oldVal))
+		}
+	}
+	output := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(newFile, []byte(output), 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// destroyUnitRetrying() destroys the unit and ensure it disappears from the
+// unit list. It could take a little time until the unit gets destroyed.
+func destroyUnitRetrying(cluster platform.Cluster, m platform.Member, serviceFile string) error {
+	maxAttempts := 3
+	found := false
+	var stdout string
+	var err error
+	for {
+		if _, _, err := cluster.Fleetctl(m, "destroy", serviceFile); err != nil {
+			return fmt.Errorf("Failed to destroy unit: %v", err)
+		}
+		stdout, _, err = cluster.Fleetctl(m, "list-units", "--no-legend")
+		if err != nil {
+			return fmt.Errorf("Failed to run list-units: %v", err)
+		}
+		if strings.TrimSpace(stdout) == "" || maxAttempts == 0 {
+			found = true
+			break
+		}
+		maxAttempts--
+	}
```

Comment:
@antrik
> Why does this run the destroy command multiple times? Shouldn't it just run the command once, and then wait for its outcome?

Good question. I tried to do run the destroy command only once, expecting it to eventually succeed. But it resulted in occasional failure. I'm not sure why, and as you might know, that's not the only unexpected error that we see. So I had to write such a hack. I'd actually love to see this hack go away.